script.js

```diff
@@ -1,12 +1,377 @@
+    }
+  });
 }
 
-window.onload = setup;
+//level 300
+
+function addEpisodesSelector() {
+  let selectAnEpisode = document.createElement("select");
+  selectAnEpisode.setAttribute("id", "episodes")
+
+  let pageNavigation = document.querySelector("nav")                                                      //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  pageNavigation.prepend(selectAnEpisode);
+
+  let getEpisodesTitles = document.getElementsByClassName("episodeName")
+
+  let episodeOption = document.createElement("option");
+  episodeOption.innerText = "All Results";
+  selectAnEpisode.appendChild(episodeOption);
+
+  for (let titleIndex = 0; titleIndex < getEpisodesTitles.length; titleIndex++) {
+    episodeOption = document.createElement("option");
+    selectAnEpisode.appendChild(episodeOption);
+
+    episodeOption.innerText = getEpisodesTitles[titleIndex].innerText;
+  }
+
+  selectAnEpisode.addEventListener("change", selectedEpisode)
+}
+
+function selectedEpisode(e) {
+  let selectedOption = e.target.value;
+  let getResultsOnPage = document.getElementsByClassName("wrapper");                             //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  let allResultsOnPage = Array.from(getResultsOnPage);
+
+  allResultsOnPage.forEach(result => {
+    if (result.innerText.includes(selectedOption)) {
+      result.classList.remove("hidden");
+    } else {
+      result.classList.add("hidden");
+    }
+
+    if (selectedOption === "All Results") {
+      result.classList.remove("hidden");
+    }
+
+  })
+}
+
+//Level 400
+
+function addShowsSelector() {
+  let selectAShow = document.createElement("select");
+  selectAShow.setAttribute("id", "shows");
+
+  let chooseEpisodeAndShow = document.createElement("label");
+  chooseEpisodeAndShow.innerText = "Choose a show";
+  chooseEpisodeAndShow.classList.add("chooseLabel")
+
+  let pageNavigation = document.querySelector("nav")
+  pageNavigation.prepend(chooseEpisodeAndShow, selectAShow);
+
+  let showsList = getAllShows();
+
+  let sortedShowsList = showsList.sort(function compare(showA, showB) {
+    if (showA.name.toUpperCase() < showB.name.toLocaleUpperCase()) { return -1 }
+    if (showA.name.toUpperCase() > showB.name.toUpperCase()) { return 1 }
+    return 0;
+  })
+
+  for (let showIndex = 0; showIndex < sortedShowsList.length; showIndex++) {
+    let showOption = document.createElement("option");
+    selectAShow.appendChild(showOption);
+
+    showOption.innerText = showsList[showIndex].name
+    showOption.setAttribute("value", `${showsList[showIndex].id}`)
+  }
+
+  selectAShow.addEventListener("change", selectedShow)
+}
+
+function selectedShow(e) {
+  showID = +e.target.value
+
+  removeCurrentPage()
```

Comment:
One thing I noticed on your page, if a select a show, the dropdown menu clears out and doesn't display which show I selected. It just shows the default (24).

This is happening because you rebuild the dropdown from scratch after clearing the page. Luckily, you do keep track of what show has been selected in a global variable `showID`. Can you think of how you could change the code so that if I have selected a show, it shows that in the `shows` drop-down selector?